File: src/cmcandy/C_Language_Answers/_0011.cpp
#include <iostream>
#include <vector>
#include <math.h>
using namespace ::std;
class Solution
{
public:
    int maxArea(vector<int> &height)
    {
        int left = 0, right = height.size() - 1;
        int area = 0;

        while(left<right){
            // area = max(area,min(height[right],height[left])*(right-left));
            int s = height[right]<height[left]?height[right]:height[left];
            int newarea = s*(right-left);
            area = area>newarea?area:newarea;
            if (height[left]<height[right])
            {
                left++;
            }
            else{
                right--;
            }
        }
        return area;
    }
};

File: src/cmcandy/Python_language_Answers/_0011.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        length = len(height)
        right = length - 1
        left = 0

        res = (right - left) * min(height[left], height[right])

        while right > left:
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
            res = max(res, (right - left) * min(height[left], height[right]))
        return res
